### ISSUE TITLE
Feed Dino 2.0

### ISSUE THREAD

**Issue by luketlancaster** [nss-evening-cohort-12/dino-kennel#9]
# Feed Dino 2.0
## User Story
As a user, in order to heal my Dino, I want the option to feed them food.  There should be options for food based on its "size" that will heal a certain amount of hit points.  The food could also be relevant to the diet of the dinosaur (IE: meat for carnivores and vegetables for herbivores)
## Dev Notes
- [] Foods Array with available foods
  - name
  - type (meat, plant/vegetable, dino chow)
  - percentage of health restored
  - food size (difference between a chicken and an elephant, or a pea and a whole tree or whatever)
### Example Foods Array
```js
foods = [
  {
    name: "chicken",
    size: "small",
    type: "meat",
    healthRestored: 0.1,
  },
  {
    name: "pineapple",
    size: "medium",
    type: "plant",
    healthRestored: 0.25,
  },
  {
    name: "elephant",
    size: "large",
    type: "meat",
    healthRestored: 0.5,
  },
  {
    name: "dino chow",
    size: "tiny",
    type: "processed",
    healthRestored: 0.05,
  }
]
```
### Required Dino Attribute Addtions
```js
dino = [
  {
    favoriteFood: TBD,
    diet: "[carnivore|herbivore|omnivore]",
    size: "[small|medium|large]",
    currentHealth: 0-100,
  }
]
```
- Feed dinosaur button
  - pops up modal window with food type/size/etc options to pick from when button is clicked
  - Based on the dinosaur size, some options should be disabled
    - i.e. small dinosaurs can't eat large food items (no tiny baby dino eating elephants!)
### STRETCH GOALS
- Dinosaur favorite foods - STRETCH GOAL
  - Established after random amount of food is eaten
    - after that amount, health benefits are increased by a certain amount (double, triple, random increase)
- Hunger level - STRETCH GOAL
- Long-term boosts after eating certain foods? - STRETCH GOAL
- Foods cooked or not? - STRETCH GOAL
- Foods based on dino size
  - i.e. small dinosaurs can't eat large items
  - large dinosaurs can eat small ones, but it takes lot more - STRETCH GOAL (would modify food health percentage before applying based on dino size)
---
### Functionality handled above already
- Foods give certain amount of health
- Food Items List
  - Chicken (small)
  - Cow (medium)
  - Elephant (large)
  - Leaves (small)
  - Pineapple (medium)
  - Pumpkins (large)
  - Dino chow (static percentage)
- Food percentages
  - dino chow - 5% total health
  - small foods - 10% total health
  - medium foods - 25% total health
  - large foods - 50% total health
- Foods based on dinosaur diet
  - carnivore
  - herbivore
  - dino chow
    - just a general average food, not better than the proper food type

**Comment by nswalters** [comment]
This is blocked by #2.

**Comment by luketlancaster** [comment]
I'm gonna think about the scope of this, maybe adjust it down to make it so we can get everything done in one day 🤔
